ebu_tt_live/twisted/node.py

```diff
@@ -19,7 +19,9 @@ def __init__(self, consumer, custom_producer):
         self._consumer.registerProducer(self, False)
         self._custom_producer.register_twisted_producer(self)
 
-    def emit_data(self, channel, data):
+    def emit_data(self, channel, data, delay=None):
+        if delay is not None:
+            reactor.callLater(delay, self._consumer.write, channel, data)
         self._consumer.write(channel, data)
 
```

Comment:
This is where unittesting just does not cut it. There was no way you could have tested this with the approach I was suggesting... you should have been able to establish how many calls were made to self._consumer.write, which is 2 n the case of delay which is wrong. This is when twisted's trial comes into play. That is their testing framework to get the reactor to tick in a defined manner. https://twistedmatrix.com/documents/16.4.0/core/howto/testing.html

In particular:

If your code uses Deferreds or depends on the reactor running, you can return a Deferred from your test method, setUp, or tearDown and Trial will do the right thing. That is, it will run the reactor for you until the Deferred has triggered and its callbacks have been run. Don’t use reactor.run() , reactor.stop() , reactor.crash() or reactor.iterate() in your tests.